lib/api/unittest/CDataFrameAnalyzerTest.cc

```diff
@@ -995,8 +994,11 @@ void CDataFrameAnalyzerTest::testRunBoostedTreeRegressionTrainingWithStateRecove
                                   params.s_FeatureBagFraction, &persisterSupplier);
         };
 
-        finalIteration = params.numberUnset() * numberRoundsPerHyperparameter - 1;
-        rng.generateUniformSamples(0, finalIteration - 1, 3, intermediateIterations);
+        finalIteration = params.numberUnset() * numberRoundsPerHyperparameter;
+        finalIteration = std::max(finalIteration, std::size_t{1}) - 1;
```

Comment:
```suggestion
        finalIteration = std::max(params.numberUnset() * numberRoundsPerHyperparameter - 1, std::size_t{0});
```

Reply:
nit: I would have only one-step assignment for `finalIteration` since the logic is not too complicated.